### PR TITLE
Fix Profile Extender not saving field names containing non-ASCII characters

### PR DESCRIPTION
**Pull request by DaazKu**
Fix https://github.com/vanilla/vanilla/issues/4103 .

The problem was that the generated slug to index the field was empty since all characters were stripped.

### CODE REVIEW

**Comment by linc** [comment]
When fixing an issue with the Backport label, remember to add it to the PR also.

File: plugins/ProfileExtender/class.profileextender.plugin.php [hunk 1]
```diff
@@ -318,7 +318,7 @@ public function settingsController_profileFieldAddEdit_create($Sender, $Args) {
                 // Make unique name from label for new fields
                 $Name = $TestSlug = substr(preg_replace('`[^0-9a-zA-Z]`', '', val('Label', $FormPostValues)), 0, 50);
                 $i = 1;
-                while (array_key_exists($Name, $Fields) || in_array($Name, $this->ReservedNames)) {
+                while (empty($Name) || array_key_exists($Name, $Fields) || in_array($Name, $this->ReservedNames)) {
```
Comment (linc) [inline plugins/ProfileExtender/class.profileextender.plugin.php]: If you have an empty `$Name` that means `$TestSlug` is also empty, which means on the next line you are allowing it to just be a numeric `$Name`, no? That seems like a problem.

**Comment by linc** [comment]
Wouldn't it have been easier to update the regex to allow Unicode?

**Comment by DaazKu** [comment]
Having the slug being only a number works perfectly.

I could allow Unicode characters but I will need to test that change more thoroughly.

**Comment by DaazKu** [comment]
`-_-` as label would still fail.

I'll make something more robust.

**Pull request merged**